gadget/gadget.go

```diff
@@ -937,32 +937,49 @@ func IsCompatible(current, new *Info) error {
 	return nil
 }
 
-// LaidOutVolumeFromGadget takes a gadget rootdir and lays out the
-// partitions as specified.
-func LaidOutVolumeFromGadget(gadgetRoot string, model Model) (*LaidOutVolume, error) {
-	info, err := ReadInfo(gadgetRoot, model)
+// LaidOutSystemVolumeFromGadget takes a gadget rootdir and lays out the
+// partitions as specified. It returns one specific volume, which is the volume
+// on which system-* roles/partitions exist, all other volumes are assumed to
+// already be flashed and managed separately at image build/flash time, while
+// the ubuntu-* roles can be manipulated on the returned volume during install
```

Comment:
```suggestion
// the system-* roles can be manipulated on the returned volume during install
```

Reply:
FWIW, this can be done in a followup

Reply:
writing comments is hard, but I will do this in a followup since we are for the moment all green :-)

Reply:
followup is https://github.com/snapcore/snapd/pull/9864